### ISSUE TITLE
Translation of posts is great! But...

### ISSUE THREAD

**Issue by adRn-s** [suvam0451/dhaaga#105]
I've seen people adding a link to a license in their posts to avoid or signal crawlers that their message should not be part of LLM training.

 (regardless of our adherence to their stance)

Shouldn't this client respect their decision? Perhaps take such posts as  untranslatable. Or add a setting, enabled by default, to respect their decision.

**Comment by suvam0451** [comment]
@adRn-s Good point. I have put a lot of consideration introducing AI to this project.

## Personal Stance

I totally understand the ethical conerns around this topic. And `I never plan on feeding image content into any irresponsibley trained AI` in this project.

The text content, however, is where I stand to differ.

I grew up in an era of open internet and digital freedom *(without the corpo stuff and crawlers)*. And, I am `really passionate about solving the language barrier issue in internet communication`.

The features I built for this project can not only translate the text content, but add an entire dimension to the interaction, and help us learn new cultural references.

In the future, I plan on adding features to help solve bidirectional chat interactions, as well.

It's something simply not possible using machine translation or DeepL today.

<img width="480" alt="Screenshot 2024-08-02 at 2 19 38 PM" src="https://github.com/user-attachments/assets/54018a42-3723-4ae0-a76c-5a6fb5731058">

<img width="480" alt="Screenshot 2024-08-02 at 2 20 37 PM" src="https://github.com/user-attachments/assets/6eee70b2-6d56-4e96-b98d-d021582c9462">

## Project Stance

Of course, views are divided on this topic:

- `The Core Project (Dhaaga)` promotes usage of AI to overcome language challenges.
- `The Libre Edition (Dhaaga-Lite)` commits to not using any AI feature in it's development. <-- This is available via [IzzyOnDroid](https://apt.izzysoft.de/fdroid/index/apk/io.suvam.dhaaga.lite)

The project license being AGPL, the responsibility/choice of using this feature falls entirely on the end-user.

Best I can do is disable this feature by default. The ultimate choice should be left to the end user.

## Common Ground and Solutions

> Shouldn't this client respect their decision?

Hmm, that is not upto the software. If the end-user is dead-set on translating your post, they will do it one way or another.

As for the person building the software, I have expressed my motivations and moral boundaries for adding these features.

^ However, I am open to feedback, criticism and solutions especially.
 
> Or add a setting, enabled by default, to respect their decision.

If you install the lite edition, it does not endorse AI features.
For the core edition, I will consider making this feature disabled by default (and add ethical warning when enabling) 🙏

> Perhaps take such posts as untranslatable.

Happy to implement. But, it's not part of fediverse spec yet (and not in horizon any time soon).

**Comment by astro-ray** [comment]
I don't think people particularly have a problem with AI translation, they have problem with OpenAI integrated into the app. Also, people are rightfully skeptical of the company. I don't think people would be as apprehensive if you used something like Libretranslate. Also, many instances support translation, is the app using it when available?

**Comment by suvam0451** [comment]
@astro-ray The lite edition does not have any AI integration.

As for the core edition, only the `long press` invokes the Ai translation. The `short press` would still call your instance DeepL, or Libretranslate.

Sometimes, you need an "explanation", rather than "translation". It is a situational tool.

^ Haven't gotten around to adding those (DeepL, LibreTranslate) in yet (some instances dont offer them. Some posts cannot be translated (e.g. - direct mentions). Still figuring it out)
